pkg/apis/flows/v1beta1/parallel_lifecycle_test.go

```diff
@@ -0,0 +1,434 @@
+				t.Errorf("unexpected condition (-want, +got) = %v", diff)
+			}
+		})
+	}
+}
+
+func TestParallelInitializeConditions(t *testing.T) {
+	tests := []struct {
+		name string
+		ts   *ParallelStatus
+		want *ParallelStatus
+	}{{
+		name: "empty",
+		ts:   &ParallelStatus{},
+		want: &ParallelStatus{
+			Status: duckv1.Status{
+				Conditions: []apis.Condition{{
+					Type:   ParallelConditionAddressable,
+					Status: corev1.ConditionUnknown,
+				}, {
+					Type:   ParallelConditionChannelsReady,
+					Status: corev1.ConditionUnknown,
+				}, {
+					Type:   ParallelConditionReady,
+					Status: corev1.ConditionUnknown,
+				}, {
+					Type:   ParallelConditionSubscriptionsReady,
+					Status: corev1.ConditionUnknown,
+				}},
+			},
+		},
+	}, {
+		name: "one false",
+		ts: &ParallelStatus{
+			Status: duckv1.Status{
+				Conditions: []apis.Condition{{
+					Type:   ParallelConditionChannelsReady,
+					Status: corev1.ConditionFalse,
+				}},
+			},
+		},
+		want: &ParallelStatus{
+			Status: duckv1.Status{
+				Conditions: []apis.Condition{{
+					Type:   ParallelConditionAddressable,
+					Status: corev1.ConditionUnknown,
+				}, {
+					Type:   ParallelConditionChannelsReady,
+					Status: corev1.ConditionFalse,
+				}, {
+					Type:   ParallelConditionReady,
+					Status: corev1.ConditionUnknown,
+				}, {
+					Type:   ParallelConditionSubscriptionsReady,
+					Status: corev1.ConditionUnknown,
+				}},
+			},
+		},
+	}, {
+		name: "one true",
+		ts: &ParallelStatus{
+			Status: duckv1.Status{
+				Conditions: []apis.Condition{{
+					Type:   ParallelConditionSubscriptionsReady,
+					Status: corev1.ConditionTrue,
+				}},
+			},
+		},
+		want: &ParallelStatus{
+			Status: duckv1.Status{
+				Conditions: []apis.Condition{{
+					Type:   ParallelConditionAddressable,
+					Status: corev1.ConditionUnknown,
+				}, {
+					Type:   ParallelConditionChannelsReady,
+					Status: corev1.ConditionUnknown,
+				}, {
+					Type:   ParallelConditionReady,
+					Status: corev1.ConditionUnknown,
+				}, {
+					Type:   ParallelConditionSubscriptionsReady,
+					Status: corev1.ConditionTrue,
+				}},
+			},
+		},
+	}}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			test.ts.InitializeConditions()
+			if diff := cmp.Diff(test.want, test.ts, ignoreAllButTypeAndStatus); diff != "" {
+				t.Errorf("unexpected conditions (-want, +got) = %v", diff)
+			}
+		})
+	}
+}
+
+func TestParallelPropagateSubscriptionStatuses(t *testing.T) {
+	tests := []struct {
+		name  string
+		fsubs []*messagingv1beta1.Subscription
+		subs  []*messagingv1beta1.Subscription
+		want  corev1.ConditionStatus
+	}{{
+		name:  "empty",
+		fsubs: []*messagingv1beta1.Subscription{},
+		subs:  []*messagingv1beta1.Subscription{},
+		want:  corev1.ConditionFalse,
+	}, {
+		name: "empty status",
+		fsubs: []*messagingv1beta1.Subscription{{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "messaging.knative.dev/v1alpha1",
```

Comment:
any particular reason we're using `messagingv1beta1.Subscription` elsewhere but `v1alpha1` within the struct?